static/app/components/searchSyntax/parser.tsx

```diff
@@ -955,41 +937,173 @@ function numeric(input: string) {
   }
   return number;
 }
-function parseDuration(input: string, _unit: string): number {
-  return numeric(input);
+
+function parseDuration(
+  input: string,
+  unit: 'ms' | 's' | 'min' | 'm' | 'hr' | 'h' | 'day' | 'd' | 'wk' | 'w'
+): {value: number} {
+  let number = numeric(input);
+
+  switch (unit) {
+    case 'ms':
+      break;
+    case 's':
+      number *= 1e3;
+      break;
+    case 'min':
+    case 'm':
+      number *= 1e3 * 60;
+      break;
+    case 'hr':
+    case 'h':
+      number *= 1e3 * 60 * 60;
+      break;
+    case 'day':
+    case 'd':
+      number *= 1e3 * 60 * 60 * 24;
+      break;
+    case 'wk':
+    case 'w':
+      number *= 1e3 * 60 * 60 * 24 * 7;
+      break;
+    default:
+      throw new Error('Invalid unit');
+  }
+
+  return {
+    value: number,
+  };
 }
-function parseNumber(input: string, unit: 'k' | 'm' | 'b' | 'K' | 'M' | 'B') {
-  const number = numeric(input);
+function parseNumber(
+  input: string,
+  unit: 'k' | 'm' | 'b' | 'K' | 'M' | 'B'
+): {value: number} {
+  let number = numeric(input);
 
   switch (unit) {
     case 'K':
     case 'k':
-      return number * 1e3;
+      number = number * 1e3;
+      break;
     case 'M':
     case 'm':
-      return number * 1e6;
+      number = number * 1e6;
+      break;
     case 'B':
     case 'b':
-      return number * 1e9;
+      number = number * 1e9;
+      break;
     case null:
     case undefined:
-      return number;
+      break;
     default:
       throw new Error('Invalid unit');
   }
+
+  return {value: number};
 }
-function parseSize(input: string, _unit: string) {
-  return numeric(input);
+function parseSize(
+  input: string,
+  // warning: size units are case insensitive, this type is incomplete
+  unit:
+    | 'bit'
+    | 'nb'
+    | 'bytes'
+    | 'kb'
+    | 'mb'
+    | 'gb'
+    | 'tb'
+    | 'pb'
+    | 'eb'
+    | 'zb'
+    | 'yb'
+    | 'kib'
+    | 'mib'
+    | 'gib'
+    | 'tib'
+    | 'pib'
+    | 'eib'
+    | 'zib'
+    | 'yib'
```

Comment:
Are we using this type in multiple places? should it be pulled up?

Reply:
its just in two places. Tbh the better way to type here might be "string" and then just make sure we handle it

Reply:
sgtm